contracts/external/cw-payroll-factory/src/msg.rs

```diff
@@ -0,0 +1,84 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cw20::Cw20ReceiveMsg;
+use cw_ownable::cw_ownable;
+use cw_vesting::msg::InstantiateMsg as PayrollInstantiateMsg;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub owner: Option<String>,
+}
+
+#[cw_ownable]
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// Instantiates a new vesting contract that is funded by a cw20 token.
+    Receive(Cw20ReceiveMsg),
+    /// Instantiates a new vesting contract that is funded by a native token.
+    InstantiateNativePayrollContract {
+        instantiate_msg: PayrollInstantiateMsg,
+        code_id: u64,
```

Comment:
how about specifying the code ID at instantiation time and allowing the owner to update it? that would allow for some social consensus over what contracts are being used.

an example of this potentially causing problems is the rentrancy issue in here from the temp singleton. using that, a malicious contract could set it's instantiator to itself, or another address it can execute messages on behalf of. if we have some social control over the code ID, we close off that type of attack a little.

Reply:
Oh, GREAT IDEA!.